app/models/calculator/weight.rb

```diff
@@ -54,6 +54,8 @@ def weight_per_final_weight_volume(line_item)
     #    Customer ends up getting 350mL (line_item.final_weight_volume) of wine
     #      that represent 2.8 (quantity_implied_in_final_weight_volume) glasses of wine
     def quantity_implied_in_final_weight_volume(line_item)
+      return 0 if line_item.variant.unit_value.zero? # Don't divide by zero ;)
```

Comment:
you could default to 1 here, that would mean the variant.weight field would be used. this way the weight will be zero even if the weight field on the variant is defined.

Reply:
no, better, I think we should default to line_item.quantity.
it would read "if unit_value is zero, the quantity_implied_in_final_weight_volume cannot be calculated so we use the line_item.quantity

Reply:
We're checking for the presence of `final_weight_volume`, but it always appears to be present and defaults to zero. I don't think we should use the volume calculations at all when `variant_unit` is "items"...?

Reply:
It still returns zero anyway, because the calculator sets `:preferred_per_kg` to `0.0`, and we multiply the result by that... :see_no_evil:

Reply:
final_weight_volume is not always present, for example for subscriptions. the check fixed a bug if you look into the commit history.

it's a good question: I think you can change the final weight volume of line items even if the variant_unit is items... in these cases final_weight_volume  becomes variant.unit_value time line_item quantity.
So, 
product.variant_unit items
variant.display_name "5 roses"
variant.unit_value 5
variant.weight 20g
line_item.quantity 2
line_item.final_weight_volume 10 (variant.unit_value * line_item.quantity)

for this line item weight calculator should calculate 40grams

but if user changes line_item.final_weight_volume from 10 to 8 (8 roses were actually delivered) the weight calculator will compute correctly 32grams :+1:  20 * (8/5)

So, I think it's good as is.

Reply:
the field name should not be final_weight_volume
as I demonstrate above it's actually final_weight_volume_units, in this case final_units.